### ISSUE TITLE
Layout graphql query data not being passed down to 404 page

### ISSUE THREAD

**Issue by baseten** [gatsbyjs/gatsby#3504]
## Description

I have a graphql query in my layout, which is passing down data to all pages. The 404 page receives this data in development, but not in production.

### Environment

Gatsby version: 1.9.52
Node.js version: 8.9.1
Operating System: Mac OS 10.11.6

### Actual result

404 page not receiving layout graphql query data

### Expected behavior

404 page should receive layout graphql query data as per other pages

### Steps to reproduce

1\. Add graphql query to `src/layouts/index.js`

2\. Add `src/pages/404.js` component using data from this query

3\. Run `gatsby build` followed by `gatsby serve` and navigate to non-existent page

**Comment by KyleAMathews** [comment]
Yeah, looks like we're not passing props into the 404 page here https://github.com/gatsbyjs/gatsby/blob/3945d0132d8b3caa32940a2749ab4809bc13d310/packages/gatsby/cache-dir/production-app.js#L168

Could you try adding `...props` there and see if it fixes things for you? And if it does, PR the fix?

Check out the contribution instructions for setting up a local dev environment https://www.gatsbyjs.org/docs/how-to-contribute/#contributing

**Comment by nikolaik** [comment]
I ran in to the same issue and adding `...props` did not do the trick as suggested by @KyleAMathews. I'm not too familiar with the code base, but looking at the surrounding code it seems that `layoutProps` is not set for 404 pages, is it not rendered as a child of a layout?
